src/main/java/org/mitre/inferno/Version.java

```diff
@@ -0,0 +1,24 @@
+package org.mitre.inferno;
+
+import java.io.IOException;
+import java.util.Properties;
+
+public class Version {
+  private static final String version;
+
+  static {
+    String temp_version;
+    try {
+      Properties properties = new Properties();
+      properties.load(Version.class.getClassLoader().getResourceAsStream("version.properties"));
+      temp_version = properties.getProperty("version");;
+    } catch (IOException e) {
+      temp_version = null;
+    }
+    version = temp_version;
+  }
```

Comment:
I hadn't seen or used a static block before either 😜 !

1. I had to use a temp variable (which my IDE recommended) because otherwise the compilation fails with `error: variable version might already have been assigned`.  Could remove `final` from the field declaration, but I think we are expecting it to be final.
2. I'm not sure there would really be a good "default version", do you have a suggestion?  I wouldn't want to duplicate the real version number here and `null` seems as informative as anything else I can think of.  I'd rather the app be more robust and continue working without the version than `System.exit`ing too.

I do think it would be good to log the error!

3. Good catch!  Weird that checkstyle didn't pick that up as even a warning...

Also realizing I could probably shorten this and not use the temp variable because Properties should return null if the property is not found, but I think its more obvious that's what is happening if we set it explicitly.  If we end up having issues or making broader use of version it might be worth rethinking the approach.